### ISSUE TITLE
TPU finetuned model corrupts

### ISSUE THREAD

**Issue by Happylkx** [salesforce/jaxformer#19]
Hi, thank you for your awesome work! 

I would like to finetune the 350M model, using tpu remote mode on a single tpu v3-8 node. I used the config file https://github.com/salesforce/jaxformer/blob/main/config/codegen_350M_multi.json, and switched `data_train_set` to my own dataset with only ~100 texts. I finetuned the model for a few epochs and converted it to pytorch for sampling. 

However, the finetuned model seems to be corrupted and repeates garbage output such as 'TETETETE', failing to give any meaningful generations. Even though the training loss goes to around 0.01 quickly, the model cannot repeat the content in the training set. I searched hyperparameters such as lr and training steps, but the problem persists.

It seems no matter what the model is trained on, the weights get corrupted quickly. Does this mean there is a discrepency between the model you pretrained and the code you published in this repo? Could you please release the original code you trained on as well?

cc: @amerine @mgomes @awaterma @ekashida @paultyng

**Comment by glicerico** [comment]
@Happylkx I found exactly the same problem with the 350M-multi model!
I was wondering if it could be a tokenization problem, but I checked that gpt2 tokenizer gives the same tokens as CodeGen tokenizer.
Although, seeing the terrible results I get, even with weird characters:

> def hello_world():����� potency 2 :op1 (sudanghai" :op1 (sudanghai" :op1 (sudanghai" :op1 (suds promoted-of (suds00200000) :op1 (suds00200000) :op1 (suds00200000) :op1 (suds00200000) :op1 (suds00200000) :op1 (suds00200000) :op1 (s Celticenos 840) :op1 (s Celtic OMGPRU party AMR: The number of the 191 GAM4 (800) :op1 (s Celtic OMGPR

it feels like the model is corrupted indeed
